deploy/templates/nstemplatetiers/test/ns_dev.yaml

```diff
@@ -13,6 +13,61 @@ objects:
     labels:
       name: ${USERNAME}-dev
     name: ${USERNAME}-dev
+
+  # Quotas and default limits for not-terminating containers (regular long-running containers)
+  # and terminating (short-lived containers like build) containers
+- apiVersion: v1
+  kind: ResourceQuota
+  metadata:
+    name: compute-deploy
+    namespace: ${USERNAME}-dev
+  spec:
+    scopes:
+      - NotTerminating
+    hard:
+      limits.cpu: "20"
+      limits.memory: ${MEMORY_LIMIT}
+      requests.cpu: 1750m
+      requests.memory: ${MEMORY_REQUEST}
+- apiVersion: v1
+  kind: ResourceQuota
+  metadata:
+    name: compute-build
+    namespace: ${USERNAME}-dev
+  spec:
+    scopes:
+      - Terminating
+    hard:
+      limits.cpu: "20"
+      limits.memory: ${MEMORY_BUILD_LIMIT}
+      requests.cpu: "2"
+      requests.memory: ${MEMORY_BUILD_REQUEST}
+- apiVersion: v1
+  kind: ResourceQuota
+  metadata:
+    name: storage
+    namespace: ${USERNAME}-dev
+  spec:
+    hard:
+      limits.ephemeral-storage: 15Gi
+      requests.storage: 15Gi
+      requests.ephemeral-storage: 15Gi
+      count/persistentvolumeclaims: "5"
+- apiVersion: v1
+  kind: LimitRange
+  metadata:
+    name: resource-limits
+    namespace: ${USERNAME}-dev
+  spec:
+    limits:
+      - type: "Container"
+        default:
+          cpu: 1000m
+          memory: 3000Mi
```

Comment:
You are increasing the default limit by 4 times (750 *4 = 3000) but the overall max limit is being increased by about 2 (7Gi vs 13Gi) plus some extra for builds (10Gi).
Even if we combine terminating and not-terminating 13+10=23Gi it's still fewer containers with the default limits:

7Gi / 750Mi = ~9.3 containers
vs.
23Gi / 3Gi = 7 containers

It means that despite the fact that we are increasing the quota the vast majority of users will actually be able to run fewer pods.
Ideally I would lower the default limit in the limitrange to balance it. 3Gi looks too high. But if you really need so high default limits then I think we have to increase the quota even higher.

Reply:
Yes, this was the minimum to get the workshop (3 pods currently running + rollout extra pod + builds) to run without setting requests/limits on those applications.  Unfortunately, the tensorflow model services were at 2500mb, so 3Gi was just a small amount of headroom.  If we want to lower the defaults and just raise the overall quota, I'm fine with that, but the workshops will have to instruct the user to set the limits/requests on those models (which they currently do).

Reply:
We are changing our serving methods in a future release in November/December, as well.